lib/rspec/rabl/attribute_matcher.rb

```diff
@@ -11,12 +11,25 @@ def description
       end
 
       def failure_message
-        "expected #{expected_value.inspect} in #{attribute_path}\n  got #{rendered_value.inspect}"
+        if nil_value?
+          "nil value when testing.\n expected #{expected_value.inspect} in #{attribute_path}\n  got #{rendered_value.inspect}\n If you want to test for a nil value, please use `with_value(nil)`"
+        else
+          "expected #{expected_value.inspect} in #{attribute_path}\n  got #{rendered_value.inspect}"
+        end
       end
 
       def matches?(subject)
         @subject = subject
-        attribute_rendered? && (!expected_value_set || rendered_value == expected_value)
+        attribute_rendered? && (!expected_value_set || rendered_value == expected_value) && nil_value_check?
+      end
+
+      def nil_value?
```

Comment:
I intended this method to check whether or not the rendered or expected value is nil. This method is only called from the `nil_value_check?` method if the user didn't set an expected value. I'm certainly open to renaming the method.